rotkehlchen/chain/ethereum/modules/eth2/decoder.py

```diff
@@ -121,62 +127,110 @@ def _decode_eth2_deposit_event(self, context: DecoderContext) -> DecodingOutput:
         )
         return DEFAULT_DECODING_OUTPUT
 
-    def _decode_eth2_consolidation_request(self, context: DecoderContext) -> DecodingOutput:
-        """Decode a request to consolidate validators.
-        Handles two types of requests:
-        - Self consolidation - changes the withdrawal credentials to 0x02 (accumulating validator)
-          See https://epf.wiki/?ref=blog.obol.org#/wiki/pectra-faq?id=q-i-have-a-validator-with-0x01-credentials-how-do-i-move-to-0x02
-        - Normal consolidation - consolidates the source validator into the target validator
+    def _decode_eth2_request_with_fee(
+            self,
+            context: DecoderContext,
+            expected_data_length: Literal[76, 116],
+    ) -> tuple[EvmEvent, EvmEvent] | None:
+        """Decode a consolidation/withdrawal request with its fee.
+        Returns the fee event and info event in a tuple or None on error.
         """
-        if len(context.tx_log.data) != 116:  # This an anonymous tx log so can't check topic[0]
+        if len(context.tx_log.data) != expected_data_length:  # This an anonymous tx log so can't check topic[0]  # noqa: E501
             log.warning(
-                f'Encountered ETH2 consolidation request with unexpected '
+                f'Encountered ETH2 request with unexpected '
                 f'data length in transaction {context.transaction}',
             )
-            return DEFAULT_DECODING_OUTPUT
+            return None
 
+        fee_event = None
         for event in context.decoded_events:
             if (
-                event.address == CONSOLIDATION_REQUEST_CONTRACT and
+                event.address == context.tx_log.address and
                 event.event_type == HistoryEventType.SPEND and
                 event.event_subtype == HistoryEventSubType.NONE and
                 event.asset == A_ETH
             ):
                 event.event_subtype = HistoryEventSubType.FEE
                 event.counterparty = CPT_ETH2
-                event.notes = f'Spend {event.amount} ETH as validator consolidation fee'
+                fee_event = event
+                break
 
-        caller_address = bytes_to_address(b'\x00' * 12 + context.tx_log.data[:20])
-        source_public_key = Eth2PubKey(bytes_to_hexstr(context.tx_log.data[20:68]))
-        target_public_key = Eth2PubKey(bytes_to_hexstr(context.tx_log.data[68:116]))
-        if source_public_key == target_public_key:
-            target = self._get_indexes_or_public_keys(public_keys=[target_public_key])[0]
-            notes = f'Request to convert validator {target} into an accumulating validator'
-        else:
-            source, target = self._get_indexes_or_public_keys(
-                public_keys=[source_public_key, target_public_key],
-            )
-            notes = f'Request to consolidate validator {source} into {target}'
+        if fee_event is None:
```

Comment:
you can get rid of the `if` check here and for a for/else.

Also change it to error, not warning.